opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpanSource.java

```diff
@@ -36,7 +34,12 @@ public ThreadLocalActiveSpan activeSpan() {
 
     @Override
     public ActiveSpan makeActive(Span span) {
-        return new ThreadLocalActiveSpan(this, span, new AtomicInteger(1));
+        return new ThreadLocalActiveSpan(this, span, new AutoFinisher());
```

Comment:
@yurishkuro so far we haven't had any problems with refcounting in any instrumentation lib. 

@mabn made a good comment https://github.com/opentracing/opentracing-java/pull/166#issuecomment-318010421. Postponing `finish` due to refcounting is nice in async situations but these use cases are mostly rare. Important is also that it can cause weird timing results if continuation of server span is passed to an async background task (but I haven't had that problem so far).